Calculator/Calculator/Program.cs

```diff
@@ -0,0 +1,267 @@
+                                var result = dateTime + timeSpan;
+                                Console.WriteLine($"Result = {result}");
+                            }
+                            // DateTime - TimeSpan
+                            else if (operatorType == OperatorType.Subtraction)
+                            {
+                                validExpression = true;
+                                var result = dateTime - timeSpan;
+                                Console.WriteLine($"Result = {result}");
+                            }
+                        }
+                    }
+                }
+
+                if (!validExpression)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine("ERROR: Invalid expression");
+                    Console.WriteLine("=================================");
+                    Console.WriteLine("These operations are supported:");
+                    Console.WriteLine("=> Arithmetic operations on Numbers");
+                    Console.WriteLine("=> String + String");
+                    Console.WriteLine("=> String - String");
+                    Console.WriteLine("=> DateTime - DateTime");
+                    Console.WriteLine("=> DateTime + TimeSpan");
+                    Console.WriteLine("=> DateTime - TimeSpan");
+                    Console.WriteLine("=> String * Number");
+                    Console.WriteLine("=================================");
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Prints the prompt message and returns user input if it is a valid string.
+        /// </summary>
+        /// <param name="message">The prompt to display.</param>
+        /// <returns>The valid string.</returns>
+        static string GetArgument(string message)
+        {
+            bool validString = false;
+            string inputString;
+            do
+            {
+                Console.WriteLine();
+                Console.Write($"{message}: ");
+                inputString = Console.ReadLine();
+                if (string.IsNullOrEmpty(inputString))
+                {
+                    Console.WriteLine("ERROR: Argument cannot be empty!");
+                }
+                else
+                {
+                    validString = true;
+                }
+            }
+            while (!validString);
+            return inputString;
+        }
+
+        /// <summary>
+        /// Returns the type of an input argument.
+        /// </summary>
+        /// <param name="argument">The input string.</param>
+        /// <returns>The corresponding ArgumentType.</returns>
+        static ArgumentType GetArgumentType(string argument)
+        {
+            var argType = ArgumentType.String;
+            var timeSpan = new TimeSpan();
+            var dateTime = new DateTime();
+            var number = 0.0;
+            if (double.TryParse(argument, out number))
```

Comment:
This might another good place for a comment, in order to explain the order of the types here.